samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java

```diff
@@ -1,2 +1,133 @@
+        + "Say 'hi' to get started.";
+    private static final String ENGLISH_ENGLISH = "en";
+    private static final String ENGLISH_SPANISH = "es";
+    private static final String SPANISH_ENGLISH = "in";
+    private static final String SPANISH_SPANISH = "it";
+
+    private UserState userState;
+    private StatePropertyAccessor<String> languagePreference;
+
+    public MultiLingualBot(UserState withUserState) {
+        if (withUserState != null) {
+            userState = withUserState;
+        } else {
+            throw new NullPointerException("userState");
+        }
+        languagePreference = userState.createProperty("LanguagePreference");
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMembersAdded(List<ChannelAccount> membersAdded,
+                                                     TurnContext turnContext) {
+        return sendWelcomeMessage(turnContext);
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        if (isLanguageChangeRequested(turnContext.getActivity().getText())) {
+            String currentLang = turnContext.getActivity().getText().toLowerCase();
+            String lang = currentLang.equals(ENGLISH_ENGLISH) || currentLang.equals(SPANISH_ENGLISH)
+                ? ENGLISH_ENGLISH : ENGLISH_SPANISH;
+            // If the user requested a language change through the suggested actions with values "es" or "en",
+            // simply change the user's language preference in the user state.
+            // The translation middleware will catch this setting and translate both ways to the user's
+            // selected language.
+            // If Spanish was selected by the user, the reply below will actually be shown in spanish to the user.
+            languagePreference.set(turnContext, lang);
+            Activity reply = MessageFactory.text(String.format("Your current language code is: %1$s", lang));
+            turnContext.sendActivity(reply);
+            return userState.saveChanges(turnContext, false);
+        } else {
+            // Show the user the possible options for language. If the user chooses a different language
+            // than the default, then the translation middleware will pick it up from the user state and
+            // translate messages both ways, i.e. user to bot and bot to user.
+            Activity reply = MessageFactory.text("Choose your language:");
+            CardAction esAction = new CardAction();
+            esAction.setTitle("Español");
+            esAction.setType(ActionTypes.POST_BACK);
+            esAction.setValue(ENGLISH_SPANISH);
+            CardAction enAction = new CardAction();
+            enAction.setTitle("English");
+            enAction.setType(ActionTypes.POST_BACK);
+            enAction.setValue(ENGLISH_ENGLISH);
+            List<CardAction> actions = new ArrayList<>(Arrays.asList(esAction, enAction));
```

Comment:
I'm not quite sure about this, can you double-check?

Reply:
You can check it [here](https://howtodoinjava.com/java/collections/arraylist/initialize-arraylist/), anyway, there are a lot of way to initialize lists in java